### ISSUE TITLE
Suggest to use larger gradient accumulation steps instead of multi GPUs

### ISSUE THREAD

**Issue by hkunzhe** [kvablack/ddpo-pytorch#10]
In the case of the same batch size, it is recommended to use a larger number of gradient accumulation steps in a single GPU instead of multi-GPUs considering huggingface/diffusers#4046. It may lead to fluctuations in the reward.

**Comment by kvablack** [comment]
Thanks so much for pointing this out! What a terrible bug. I've been able to fix it so that gradients are synchronized properly across GPUs, but it uses more memory for some reason (up to 16GB from 10GB before the change).

**Comment by bhattg** [comment]
Hi! Does this bug effect any of the findings in the paper?

**Comment by kvablack** [comment]
@bhattg no, fortunately the results in the paper all used the [original Jax codebase](https://github.com/jannerm/ddpo).
